### PR TITLE
Improve handling of corrupt page templates data

### LINKED ISSUE #10936: Saved templates not visible
## Bug Description

Saved templates are not appearing in the editor view. The list is empty and there is no errors in the console. I noticed that the API result of saved templates has a lot of templates but there is two types of items in the list. One type has `story_data` as an object with data on it and other type is `story_data` as an array without data on it. Could the view be failing silently because of the empty `story_data` arrays? On my other environments the saved template views are working properly and there is no empty `story_data` arrays in the response.

## Expected Behaviour

Saved templates should appear in the template list just like the default templates does.

## Steps to Reproduce

I cannot provide the steps to reproduce because I don't know what happened. Since the third of February the saved template list in the editor view has been empty. Could this be related to some unsuccessful migration?

## Screenshots

Here is a screenshot of the editor view with console open showing the results of `/wp-json/web-stories/v1/web-story-page/?context=edit&per_page=100&page=1&_web_stories_envelope=true&_fields=id%2Cstory_data%2Cbody%2Cheaders&_embed=wp%3Afeaturedmedia&_locale=user` API request.
Screenshot 2022-03-16 at 10 42 55

## Additional Context

- Plugin Version: 17.0.1
- WordPress Version: 5.9.2
- Operating System: macOS Big Sur 11.6
- Browser: Chrome, Brave, Firefox

### PR DESCRIPTION
**Pull request by swissspidy**
## Context

As reported in #10936, the Saved Templates library tab is empty if there is 1 page template with empty `story_data`.

`story_data` is empty if the story data stored in `post_content_filtered` is corrupt and not valid JSON anymore.

This leads to some JS errors being thrown and silenced, causing the whole tab to be empty.

## Summary

This PR improves handling of corrupt page templates by ensuring certain object keys are always present, even if `story_data` is empty.

## To-do

In the future, if this becomes a more frequent issue, we might want to look into ways to surface this to the users.

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Create a new story with some content
2. Save as a template
3. Modify the template in the database, emptying the `post_content_filtered` column
4. Go to the editor and to the Saved Templates list
5. Verify that the all saved templates appear as expected

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

Adds new error tracking for when loading page templates fails.

### Does this PR have a legal-related impact?

No

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #10936

### CODE REVIEW

**Comment by spacedmonkey** [comment]
If the data is invalid in the database. Could we just fix this at the PHP level. If the data is invalid, either remove api or empty an empty array / object?

**Comment by swissspidy** [comment]
Mhm yeah maybe; not sure what's the best way there.

`Stories_Base_Controller` already does _some_ handling and returning `null` if it's invalid JSON. Returning an empty object would not change anything and still cause client-side issues.

So this type of hardening here is required nonetheless.

Also worth noting that `Stories_Base_Controller` is also used for stories themselves, so need to be careful not to paper over errors there.

In short, I think PHP-level changes need some thorough thinking. We can open a ticket to discuss this further.

**Review by maxyinger — APPROVED** [review]

**Review by spacedmonkey — APPROVED** [review]
